muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java

```diff
@@ -317,11 +317,8 @@ public Response listWorkspaces(
     }
     
     if (doMinVisitFilter) {
-      if (userEntity != null) {
-        workspaceEntities = workspaceVisitController.listWorkspaceEntitiesByMinVisitsOrderByLastVisit(userEntity, minVisits);
-      } else {
-        workspaceEntities = workspaceVisitController.listWorkspaceEntitiesByMinVisitsOrderByLastVisit(sessionController.getLoggedUserEntity(), minVisits);
-      }
+      UserEntity loggedUserEntity = sessionController.getLoggedUserEntity();
```

Comment:
Should returrn UNAUTHORIZED when doMinVisitFilter is active and user is not logged in,  otherwise this could lead to NPE